src/main/java/seedu/address/model/person/patient/Weight.java

```diff
@@ -0,0 +1,53 @@
+package seedu.address.model.person.patient;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Patient's weight in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidWeight(String)}
+ */
+public class Weight {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Weight should be in kilograms (kg), expressed as a positive decimal number.";
+
+    public static final String VALIDATION_REGEX = "^[0-9]+(\\.[0-9]{1,2})?$";
```

Comment:
This regex seems to fail for decimal numbers with floating point on https://regex101.com/. It works with integers though.

Reply:
Should we also check for leading and trailing whitespaces?